aws-android-sdk-core/src/main/java/com/amazonaws/http/UrlHttpClient.java

```diff
@@ -85,6 +94,38 @@ public HttpResponse execute(final HttpRequest request) throws IOException {
         return createHttpResponse(request, connection);
     }
 
+    HttpURLConnection getConnection(URL url) throws IOException {
+        if (config.getProxyHost() != null) {
+            SocketAddress socketAddress;
+            if (config.getProxyPort() != -1) {
+                socketAddress = new InetSocketAddress(config.getProxyHost(), config.getProxyPort());
+            } else {
+                socketAddress = new InetSocketAddress(config.getProxyHost(), 80);
+            }
+            proxy = new Proxy(Proxy.Type.HTTP, socketAddress);
```

Comment:
Can this proxy object be created once per instance instead of create one per request?